solutions/USACO feb20-clocktree-silver.mdx

```diff
@@ -0,0 +1,109 @@
+---
+id: usaco-1016
+source: USACO Silver 2020 February
+title: Clock Tree
+author: Melody Yu
+---
+
+[Official Analysis](http://www.usaco.org/current/data/sol_clocktree_silver_feb20.html)
+
+We can build a graph with the rooms as nodes and corridors as edges. Consider a simple case: there are 2 nodes in the graph and one edge.
+Bessie can start from one of the rooms, go to its neighbor, increase the clock, go back, and increase the starting room's clock.
+During this process of moving back and forth, notice that the difference between two rooms' clock readings may only change by 1.
+If the starting clock readings differ by one, Bessie can start from the bigger clock reading room and travel back and forth until both of the clocks point to 12.
+If the starting clock readings differ by more than one, it is impossible for Bessie to make both clocks point to 12.
+
+If there are more nodes, we can treat it similar to the two node case by considering the two groups in a bipartite partition. As all trees are bipartite graphs, this partition always exists.
+
+The first time Bessie visits a neighbor node, the sum of clocks of the first group increases by 1.
+Then Bessie visits another neighbor node, the sum of clocks of the second group increases by 1.
+As both sums increase by one, the difference between the sums does not change.
+This process repeats until all clocks reach 12.
+
+We can check the initial sum difference between the two groups (mod 12) to figure out the final clock numbers.
+
+1. If group0 is the same as group1 (mod 12), Bessie can start from any node.
```

Comment:
The last two comments here were not resolved:

https://github.com/cpinitiative/usaco-guide/pull/1474#discussion_r637164340

Reply:
oops, missed this D: @cskitty ?

Reply:
Fixed the comments, sorry about the lateness!